src/logging/logger.ts

```diff
@@ -63,8 +65,16 @@ export class ComponentLogger implements ZLUX.ComponentLogger {
     this._messages = messages;
   }
 
+  public _setMessages(table: MessageTable) {
+    this._messages = table;
+    this._subLoggers.forEach(sublogger => sublogger._messages = table);
+  }
+
   makeSublogger(componentNameSuffix:string): ComponentLogger {
-    return new ComponentLogger(this.parentLogger, this.componentName+':'+componentNameSuffix);
+    const subLogger = new ComponentLogger(this.parentLogger, this.componentName+':'+componentNameSuffix);
+    subLogger._messages = this._messages;
```

Comment:
If you created a set method for the messages, you might as well use it here too. The forEach part would just get ignored

Reply:
The second time around it wouldn't, since you can have multiple subloggers, so it would be extra processing not needed.

Reply:
Wait, but aren't you changing the messages of the new componentLogger that you just created?

so subLogger._setMessages(this._messages) should never pass over subLoggers here because you don't have subLoggers yet in this object